src/main/java/me/karakelley/http/Main.java

```diff
@@ -1,33 +1,19 @@
 package me.karakelley.http;
 
-import me.karakelley.http.filesystem.FileFinderCache;
-import me.karakelley.http.filesystem.PublicDirectory;
-import me.karakelley.http.filesystem.RealFileFinder;
-import me.karakelley.http.handlers.Application;
 import me.karakelley.http.server.*;
 import me.karakelley.http.utility.CommandLineArguments;
 import me.karakelley.http.utility.SystemExit;
 
 import java.util.Map;
-import java.util.function.BiConsumer;
 
 class Main {
   public static void main(String[] args) {
-    main(args, (argsHash1, serverConfiguration1) -> {
-      if (!argsHash1.containsValue("nodirectory")) {
-        serverConfiguration1.setHandler(new Application(PublicDirectory.create(argsHash1.get("directory"), new FileFinderCache(new RealFileFinder()))));
-      } else serverConfiguration1.setHandler(new Application());
-    });
-  }
-
-  public static void main(String[] args, BiConsumer<Map<String, String>, ServerConfiguration> configurationBiConsumer) {
     Map<String, String> argsHash = CommandLineArguments.parse(args, new SystemExit());
     ServerConfiguration serverConfiguration = new ServerConfiguration();
     serverConfiguration.setPort(argsHash.get("port"));
-
-    configurationBiConsumer.accept(argsHash, serverConfiguration);
-
+    serverConfiguration.setHandler(new ApplicationFactory().create(argsHash.get("directory")));
```

Comment:
What are the benefits of a separate class with instance methods vs adding a `static` method on `Application`?